### DISCUSSION TITLE
Design suggestions megathread

### DISCUSSION THREAD

**Discussion by henrygd** [henrygd/beszel#56] [Ideas]
Please post any ideas for design improvements in this thread. And upvote the ideas you like.

Don't worry about critiquing the current design. I'm not an expert and won't be offended. It's helpful to get tips from people who are better at it.

For reference, the app uses [shadcn/ui](https://ui.shadcn.com/), so any components / layouts from there should be easy enough to implement.

**Comment by henrygd** [comment]
To start, here are suggestions from @alex3305 in issue #31:

> Thanks for creating this nifty little tool. As you can probably tell I have been experimenting with it all day and enjoying it really well. One thing I have noticed is that the current layout has pretty low information density. This works really well on mobile, but perhaps not so on desktop. I think it would improve the project with very slight layout changes on larger screens. I have made some mockups to show what I mean:
> 
> ![image](https://private-user-images.githubusercontent.com/5155694/352621420-359853ca-8c5c-4b05-aa13-6768f3442c7c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxODI0OTIsIm5iZiI6MTcyMjE4MjE5MiwicGF0aCI6Ii81MTU1Njk0LzM1MjYyMTQyMC0zNTk4NTNjYS04YzVjLTRiMDUtYWExMy02NzY4ZjM0NDJjN2MucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjhUMTU1NjMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTAxODBhYjFjMTc5ZDFjNDhmOTdkM2Q0MDkyYmNiYjFhNjI5YmZhYmE5MTVhMmM1YTg5YjQ1ODE4ODhmNzJiZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.2dRssT98huCKCWFZeIvT31m-l_3IrPBNU-7wY9UX0EI) ![image](https://private-user-images.githubusercontent.com/5155694/352618811-471bd333-9f1e-4d3a-bbed-93e5db67036e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjIxODI0OTIsIm5iZiI6MTcyMjE4MjE5MiwicGF0aCI6Ii81MTU1Njk0LzM1MjYxODgxMS00NzFiZDMzMy05ZjFlLTRkM2EtYmJlZC05M2U1ZGI2NzAzNmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjhUMTU1NjMyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2Y4Mzg5YzU3NjI4NDU3MTYxOGRmMTE4YWFhMmY3MjJmYjE5YmY0Zjc1OGQzYzZhMjQ1YjNlZmVmNDBkMGUzMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gQL83rE85os8X-uphx3HKeQNSbqWrXLOIajQLazeFhU)

**Reply by alex3305** [reply]
Repasting the images so they are also visible here:
![image](https://github.com/user-attachments/assets/f87e0882-25e4-42a8-bce9-23879bf3f3fb)
![image](https://github.com/user-attachments/assets/0ec6b478-5b3a-4562-8679-7ed072f48731)

**Reply by henrygd** [reply]
Updated in 0.2.0. Thanks again for the suggestions!

**Comment by henrygd** [comment]
I like these changes a lot, but I'm not sure about the buttons in the top bar shifting around. So I think it might be better to have the "Add System" button on every page.

I also find it helpful to line up CPU spikes vertically with docker containers, so maybe we can make the grid configurable. A button that switches between full width and side-by-side layouts. Or otherwise have the CPU charts full width and everything else in the grid.

Any thoughts are appreciated!

**Reply by alex3305** [reply]
First of all thanks for your openness to discuss these suggestions. That is greatly appreciated 🚀. I saw your reply in #31, but will reply here with your comments inline for consistency. 

> I like these changes a lot, but I'm not sure about the buttons in the top bar shifting around. So I think it might be better to have the "Add System" button on every page.

Having the same top bar for all pages (=uniformity) was always my intention, but for my mock-ups I just moved around some elements. Doing that across pages was a bit tedious so I omitted that. I should have a been a little more clear about that though.

> For the charts, I was split on whether to go with the 2x2 grid. I agree it's much better for getting a quick overview of the system.

Both are fine. Don't worry about it. However on desktop IMHO a 2 column grid would suit better, but I guess that's also a matter of preference. 

> A button that switches between full width and side-by-side layouts.

What I would do is make two layouts:
* Single column
* Two column

I would make the single column default on mobile devices (ie. <700px) and two column on screens with more real estate. On those bigger screens I would also add a button to switch between the two. On those smaller screens I wouldn't show the button. But that's just how I should have do it. It doesn't mean that's the best way, just my preference 🤷 .

> Or otherwise have the CPU charts full width and everything else in the grid.

I wouldn't mind having the CPU chart half width. Even half chart has plenty of information density. However I mostly look at trends and previous x-minutes.

---
The only thing from the current layout that really bothers me are the Docker CPU and Docker Memory tooltips. I have about 60 containers running and the thing is borderline unusable. On the [cAdvisor dashboard](https://grafana.com/grafana/dashboards/14282-cadvisor-exporter/) that I previously used there is a table next to the graph with all the containers. Which is way more usable.

**Reply by henrygd** [reply]
I agree with pretty much all of this.

Someone suggested having a filter bar on container charts so you could search through the items in the tooltip. And possibly update the chart to show only the matching containers.

Having a little table like the cadvisor example is more functional, but where could we fit that in? I'm even open to a major redesign. I just want something that's usable and doesn't look like shit.

**Reply by utlilb** [reply]
I would love to have the ability to add/remove/rearrange columns on the main screen.

**Reply by henrygd** [reply]
I'll probably add column display selection in the next release or two.

**Comment by ctrlaltdelete007** [comment]
Thanx for the great new layout!
Is it possible to use more than one filesystem entry? md2; md3 What would be the syntax?
  FILESYSTEM: /dev/md3

**Reply by henrygd** [reply]
Not yet. Hopefully in the next release. See #36.

**Comment by Simonh2o** [comment]
Missing the single (non-split) layout already, hoping it will be customizeable in future releases!

**Reply by henrygd** [reply]
I'll add a way to toggle it.

**Comment by lephtHanded** [comment]
main page options:
current load in page is a list of all the hosts with some information for them.

could there be a user chosen main page, or link to a page that has a certain stat for ALL hosts... then you can click in to them for more information?

**scenario**:  i like looking at bandwidth, others may prefer Disk I/O or CPU usage... what ever stat you like most. 
- the page I want to see has *all* the bandwidth graphs sitting cozy to each other like the host view of all stats.  
- each graph labeled Bandwidth-[hostname]
- if there is some weird spike and i want to investigate, i click that hosts chart and boom, i'm looking at that hosts full stats.

**Reply by henrygd** [reply]
Yes, I mentioned exactly this in #29 and it's in the queue after webhook alerts and a couple other things.

**Comment by TheTaran** [comment]
Will there integration of SNMP checks? With the Agent is only possible to check Servers/Systems with a accesible OS. With SNMP integration it will be possible to integrate also Network devices in one Dashboard (Full OVerview.)

**Reply by henrygd** [reply]
I have no experience working with SNMP, so probably not. At least not in the near future.

Feel free to open an issue with more information on exactly what you'd like to see and I'll look into it further down the line.

**Reply by chaugan** [reply]
> I have no experience working with SNMP, so probably not. At least not in the near future.
> 
> Feel free to open an issue with more information on exactly what you'd like to see and I'll look into it further down the line.

@henrygd 

I have forked and started (soon finished) an SNMP agent feature to the project!


![snmp2](https://github.com/user-attachments/assets/7b1d8332-a704-4c3b-8833-de9883e8bbc0)
![snmp1](https://github.com/user-attachments/assets/080cfdc7-c7e5-4dd7-b092-30deb5ae7476)

**Comment by wwng2333** [comment]
Thanks for you work, this project is amazing. :)
I have 2 small problems here:
1.Can we add the system loadavg at the index page? that will help us know the main status of the server.
2.Chinese translate quailty is suck, is that translated by machine? I can give him some polish if required.

**Reply by henrygd** [reply]
1. Yes it's possible. Currently the CPU value is the average of the last one minute, so it does provide some larger context as opposed to being simply a moment in time snapshot.

2. Polish has been added already and will be in the next release. Chinese is machine translated currently. If you can improve it that would be a huge help: https://crowdin.com/project/beszel

**Reply by wwng2333** [reply]
> 2. Polish has been added already and will be in the next release. Chinese is machine translated currently. If you can improve it that would be a huge help: https://crowdin.com/project/beszel

OK, I did some improvement of Chinese.

**Reply by henrygd** [reply]
Thanks!

**Reply by wwng2333** [reply]
> Thanks!

More improvemnt is done. Need i notice you for that at here?

**Reply by henrygd** [reply]
Thanks! You do not need to notify me here. I will see your additions in Crowdin.

**Comment by wwng2333** [comment]
For intel iGPUs, we can indicate their gpu with `/sys/class/drm/card*/gt_cur_freq_mhz` and `/sys/kernel/debug/dri/*/i915_frequency_info`.

**Comment by wwng2333** [comment]
Some idea about hostname problem:
1. Should add check for IPv6 hostname like `[240e::aaaa]`, it not work without `[]`.
2. Support for dualstack network machine, if IPv4 or IPv6 not work, should reconnect via another one.

**Comment by hdgsn** [comment]
Just a suggestion that "Active Alerts" gets colourised in some form - perhaps a simple background color.  They currently seem to get a little lost on the page given their relative importance.

![image](https://github.com/user-attachments/assets/d92d1e0b-0604-4fab-9a15-f1296bcdea98)

**Comment by sashablue** [comment]
Hello,
adding an Uptime column in Visible Fields would also be very useful.

Cheers

**Comment by trevigol** [comment]
Hi, it would be cool to be able to choose the order of the widgets. For example I would like to have in first position the widget about CPU temperature.

also it would be great to have a widget with the processes that are using the cpu
